### PR TITLE
Add support for teams

### PR DESCRIPTION
**Pull request by builder-247**
In this PR I'll be adding support for teams.

### TODO
- [x] finish Team class implementation
- [x] pre-1.13 support
- [x] add a display method to scoreboards
- [x] tests
- [x] documentation

Related: https://github.com/PrismarineJS/mineflayer/issues/1377, https://github.com/PrismarineJS/mineflayer/issues/1714, https://github.com/PrismarineJS/mineflayer/pull/864

### CODE REVIEW

File: lib/team.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  remove (name) {
+    const removed = this.membersMap[name]
+    delete this.membersMap[name]
+    return removed
+  }
+
+  // Return a chat component with prefix + color + name + suffix
+  displayName (member) {
+    // TODO
```
Comment (builder-247) [inline lib/team.js:39]: Is there an existing method of joining multiple `ChatMessage`s?

File: lib/team.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    return this.membersMap[name]
+  }
+
+  remove (name) {
+    const removed = this.membersMap[name]
+    delete this.membersMap[name]
+    return removed
+  }
+
+  update (packet) {
```
Comment (rom1504) [inline lib/team.js:73]: could you do the packet handling in the team plugin ?
we try to keep such classes independent from the packets so they stay fairly stable
(that means taking as input directly {name, friendlyFire, ...} here and explicitely passing each of those fields in the team plugin
that will make it easier to adapt when mojang (inevitably) changes the fields

File: lib/team.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+
+module.exports = loader
+
+function loader (mcVersion) {
+  ChatMessage = require('prismarine-chat')(mcVersion)
+  MessageBuilder = ChatMessage.MessageBuilder
+  return Team
+}
+
+function colorString (color) {
```
Comment (rom1504) [inline lib/team.js:12]: could you check whether prismarine-chat could be used to do that ? I feel we already have that info there

**Review by rom1504 — CHANGES_REQUESTED** [review]
Looks good overall!
Thanks for working on this.

I put 2 comments of things to improve.

File: lib/team.js (hunk 3 above)
Comment (builder-247) [reply lib/team.js:12]: while we have the data, it's not accessible currently

**Review by rom1504 — APPROVED** [review]

**Pull request merged**